Cesium.Compiler/Compilation.cs

```diff
@@ -88,16 +88,23 @@ private static async Task GenerateCode(AssemblyContext context, string inputFile
     private static void SaveAssembly(
         AssemblyContext context,
         SystemAssemblyKind targetFrameworkKind,
-        string outputFilePath)
+        string outputFilePath,
+        string compilerRuntimeDll)
     {
         context.VerifyAndGetAssembly().Write(outputFilePath);
 
         // This part should go to Cesium.SDK eventually together with
         // runtimeconfig.json generation
-        var compilerRuntime = Path.Combine(AppContext.BaseDirectory, "Cesium.Runtime.dll");
         var outputExecutablePath = Path.GetDirectoryName(outputFilePath) ?? Environment.CurrentDirectory;
         var applicationRuntime = Path.Combine(outputExecutablePath, "Cesium.Runtime.dll");
-        File.Copy(compilerRuntime, applicationRuntime, true);
+
+        // Prevent copying of the Cesium.Runtime if compile in same directory as compiler.
+        var outputExecutablePathFullPath = GetNormalizedFullPath(outputExecutablePath);
+        var compilerPathFullPath = GetNormalizedFullPath(AppContext.BaseDirectory);
+        if (outputExecutablePathFullPath != compilerPathFullPath)
+        {
+            File.Copy(compilerRuntimeDll, applicationRuntime, true);
+        }
```

Comment:
Is this the situation when we copy the file into itself? If so, then maybe it would be better to compare `compilerRuntimeDll` and `applicationRuntime`?

As a bonus, you won't need the separator tricks to compare them, since both will be full file paths.

Reply:
Command line produce
```
Cesium.Compiler.exe C:\Cesium\Cesium.IntegrationTests\hello.c -o test.exe --modulekind console
```
have following values
```
compilerRuntime: C:\Cesium\artifacts\bin\Cesium.Compiler\Debug\net6.0\Cesium.Runtime.dll
applicationRuntime: Cesium.Runtime.dll
```
When call `Path.GetFullPath("", AppContext.BaseDirectory)` it return `C:\Cesium\artifacts\bin\Cesium.Compiler\Debug\net6.0` but `Path.GetFullPath(compilerRuntime, AppContext.BaseDirectory)` return `C:\Cesium\artifacts\bin\Cesium.Compiler\Debug\net6.0\` so I think normalization is required

Reply:
I still think it's better to compare `Path.GetFullPath(compilerRuntime)` and `Path.GetFullPath(applicationRuntime)`.

Reply:
I mean, the invariant of taking one of the artifacts from `AppContext.BaseDirectory` is very fragile. To make the code more robust, it's better to directly protect from what's happening: from copying the file into itself.